m3/include/m3/reporter.h

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
```

Comment:
should these files live under `tally-cpp/tally/m3` too?

Reply:
Since it's being built as a separate library I like keeping these files in a separate directory. It is weird though now that m3 changed the namespace from `m3` to `tally::m3`. Bazel has an `include_prefix` option so that we can add `tally` to the file path for the headers, i.e. `m3/reporter.h` -> `tally/m3/reporter`. Would that make things clearer?

Reply:
Ideally the layout of the repo, esp headers would be where people expect them to be. I think the bazel prefix thing would confuse people more then help. Can you distribute the `m3` bits as a separate dep from the same repo and have the structure be nested?

Reply:
Perhaps `reporter.h` can live in its own namespace, then `tally`, `m3` and `reporter` can all be separate.

Reply:
m3 is more user-friendly. People presumably are looking for m3 client code here.